pages/data-design/branches/pulling-pushing-items-between-branches.mdx

```diff
@@ -1,116 +1,90 @@
 
 ![Pressing push button](/images/pulling-pushing-items/press-push-button.png)
 
-When that button is clicked, you will get a modal that prompts you to pick a branch to push the changes to and review your selection.
-You have to first pick a branch so that Avo can accurately display how the changes will be applied, as it might differ per branch if the item already exists on the destination branch.
+Clicking the button opens a modal prompting selection of a destination branch and a review of the changes.
+A branch must be selected first so Avo can accurately display how the changes will be applied—this may vary depending on whether the item already exists on the selected branch.
 
 <center><img
   src="/docs/images/pulling-pushing-items/empty-push-modal.png"
   width="60%"
   alt="Empty review modal"
 /></center>
 
-When the destination branch is selected, the actual diff for the destination branch is shown. The items shown can be expanded to show the details of what will be added to the destination branch.
-If the item you selected to push is an event or a property bundle, any related property changes are displayed below. Avo determines which changes are necessary to push to the destination branch leaving out changes that are unrelated to the selected event.
-See the [property changes related to event](#property-changes-related-to-event) section for details.
-Once you have reviewed your selection, you can click the Push to branch button and the changes will be applied to the destination branch.
+After selecting a destination branch, changes can be [reviewed](#reviewing-changes-to-be-pushed-or-pulled), including which [related changes](#property-changes-related-to-event) to include in the push..  
+Once pushed, the item becomes available for use on the destination branch.
 
-<center><img
-  src="/docs/images/pulling-pushing-items/collapsed-diff-modal.png"
-  width="60%"
-  alt="Empty review modal"
-/></center>
-
-Then you can navigate to your destination branch and use the item you pushed there.
+## Pulling items from another branch
 
-![Empty review modal](/images/pulling-pushing-items/destination-branch.png)
+Events, properties, and property bundles can be pulled into the current branch during item search or creation:
+- [Item details](#item-details): When searching for items in context of another item details (for example adding a property to an event in event details)
+- [Item creation](#item-creation): When creating new events or properties (for example by clicking the "+ New Event" button in the events view)
+- [Global search](#global-search) When searching for items in cmd+k or search bar
 
-## Pulling items from another branch
+When an event from another branch is surfaced in any of the contexts listed above, selecting it initiates the pulling process.
+A modal appears with a detailed overview of the changes that will be applied, including options to review and configure related changes.
```

Comment:
First the branch preview opens up though, should we mention that?

Reply:
But not always, hmmm

Reply:
You mean the item details? There is a screenshot and workds for when that happens (item creation and global search). Is it not clear